openedx/frontend-app-course-authoring
# feat: complete subsection & unit features

**Description:**

This PR combines following features and PRs:

* https://github.com/openedx/frontend-app-course-authoring/pull/769 `Private-ref` [BB-8278](https://tasks.opencraft.com/browse/8278)
* https://github.com/openedx/frontend-app-course-authoring/pull/781 `Private-ref` [BB-8431](https://tasks.opencraft.com/browse/8431)
* https://github.com/openedx/frontend-app-course-authoring/pull/788 `Private-ref` [BB-8421](https://tasks.opencraft.com/browse/8421)
* https://github.com/open-craft/frontend-app-course-authoring/pull/27 `Private-ref` [BB-8423](https://tasks.opencraft.com/browse/8423)
* https://github.com/open-craft/frontend-app-course-authoring/pull/22 `Private-ref` [BB-8277](https://tasks.opencraft.com/browse/8277)
* https://github.com/open-craft/frontend-app-course-authoring/pull/26 `Private-ref` [BB-8430](https://tasks.opencraft.com/browse/8430)

<details>
  <summary>

feat: move up & down option `Private-ref` [BB-8422](https://tasks.opencraft.com/browse/8422)

</summary>
  
**Description:**

Adds move up and down option to sections, subsections and units menu.

**Screenshots:**

![image](https://github.com/openedx/frontend-app-course-authoring/assets/10894099/cd4144df-b3b8-4fd5-b7f7-1bd166210fea)

**Test instructions:**

* Get devstack up and running using `make {lms,cms}-up`.
* Start npm devserver in this repo using `npm start`.
* Goto http://localhost:2001/course/course-v1:edX+DemoX+Demo_Course/ and click on three dots of any section or subsection.
* It should move sections/subsections as expected.
* Options should be visible based on the position, like first section should not have an option to move up etc.

</details>

**Notes:** Please use rebase option to merge (Do not squash, as the commits are already squashed based on the features and authors).
